iroh-p2p/src/swarm.rs

```diff
@@ -104,15 +104,19 @@ pub(crate) async fn build_swarm(
         .with_max_established_incoming(Some(config.max_conns_in))
         .with_max_established_outgoing(Some(config.max_conns_out))
         .with_max_established_per_peer(Some(config.max_conns_per_peer));
-    let swarm = SwarmBuilder::new(transport, behaviour, peer_id)
+    let swarm = SwarmBuilder::with_executor(transport, behaviour, peer_id, Tokio)
         .connection_limits(limits)
         .notify_handler_buffer_size(config.notify_handler_buffer_size.try_into()?)
         .connection_event_buffer_size(config.connection_event_buffer_size)
         .dial_concurrency_factor(config.dial_concurrency_factor.try_into().unwrap())
-        .executor(Box::new(|fut| {
-            tokio::task::spawn(fut);
-        }))
         .build();
 
     Ok(swarm)
 }
+
+struct Tokio;
+impl Executor for Tokio {
+    fn exec(&self, fut: std::pin::Pin<Box<dyn futures::Future<Output = ()> + Send>>) {
+        tokio::task::spawn(fut);
```

Comment:
Is this so we can futz around with different executors / configs?

Reply:
nope, it is because the rust-libp2p api changed